src/Microsoft.Identity.Web/TokenAcquisition.cs

```diff
@@ -170,11 +171,24 @@ internal partial class TokenAcquisition : ITokenAcquisitionInternal
 
                 context.HandleCodeRedemption(null, result.IdToken);
             }
+            catch (MsalServiceException exMsal) when (IsInvalidClientCertificateError(exMsal))
+            {
+                DefaultCertificateLoader.ResetCertificates(_microsoftIdentityOptions.ClientCertificates);
```

Comment:
From what I understand .. the user configures one or more certificate descriptions. Each certificate description can be used to fetch 0 or 1 certificates. 

I think here we only deal with the case where there is a KeyVault certificate description and we try to reload the certificate, using the existing description. 

I am not sure how M.I.W. deals with multiple cert descriptions, e.g. one from KV, one from my local store? Maybe we can chat about this. 

In any case, I think there should be an explicit API on the ClientCertificate class, smth like `Reload`, which may be implemented by some cert types (keyvault) but not others.

Reply:
Descriptions from other than KV won't have several versions (KeyVault does), but this is not even a goal. 
The cert rotation is still enabled for other descriptions than KeyVault if the developer provides several cert descriptions (one that will expire soon, and another one), whatever the certificate source. The thing with KeyVault is that the same certificate source can have a newer version which extends the expiry of the previous, now expired version. the filepath description could be the same too, and probably the subject name one too from the certificate store.

Why Reload? is it an optimization?